contracts/models/contract_enterprise.py

```diff
@@ -0,0 +1,50 @@
+from odoo import api, fields, models, tools, _
+from odoo.exceptions import UserError, ValidationError
+from datetime import datetime, timedelta
+
+class ContractEnterprise(models.Model):
+    _name = "contract.enterprise"
+    _description = "contract enterprise model"
+
+    def _default_sign_date(self):
+        next_week = fields.Datetime.now() + timedelta(weeks=1)
+        if next_week.weekday() >= 5:
+            next_week += timedelta(days=7 - next_week.weekday())
+        return next_week
+
+    id = fields.Integer('id', required=True)
+    name = fields.Char("contract name", required=True)
+    start_date = fields.Datetime('start date', required=True, select=True, default=datetime.now())
+    end_date = fields.Datetime('end date', required=True, select=True)
+    type_contract = fields.Integer('contract type', required=True)
+    sign_date = fields.Datetime('sign date', required=True, select=True, default=_default_sign_date)
+    salary_level = fields.Float('salary level', required=False, default=0)
+    effective_salary = fields.Float('effective salary', required=False, default=0)
+    employee_id = fields.Many2one('employee.directory', string="Employee", required=True, ondelete='cascade')
+    '''
+        ondelete : có 4 type
+            + cascade : khi một bản ghi được xóa, tất cả các bản ghi liên quan trong table con sẽ cũng được xóa.
+            + set null :  khi một bản ghi được xóa, trường Many2one sẽ được đặt thành giá trị NULL.
+            + restrict : Odoo sẽ không cho phép xóa bản ghi nếu có bản ghi liên quan trong bảng con.
+            + no action : Odoo sẽ không thực hiện bất kỳ hành động nào khi một bản ghi được xóa. Bạn phải quản lý các hành động liên quan bằng cách thủ công
+    '''
+    sign_day = fields.Date(string="Sign Date")
+    status = fields.Selection([
+                ('new', 'New'),
+                ('running', 'Running'),
+                ('expired', 'Expired'),
+                ('pause', 'Pause')
+            ], string='Status', default='new')
+
+    total_salary = fields.Float(string='Total Salary', compute='_compute_total_salary', store=True)
```

Comment:
store dùng khi nào?

Reply:
giá trị default của store = False, store được sử dụng khi xác định có lưu giá trị trường đó vào trong DB hay không, nếu store=True --> lưu giá trị trường đó vào DB

Reply:
giá trị default của store = False ở tất cả các trường hợp hay chỉ khi dùng với compute?

Reply:
giá trị store mặc định là False cho tất cả các trường trong Odoo, bao gồm cả các trường tính toán computed và các trường thông thường ạ